system/jlib/jptree.ipp

```diff
@@ -693,21 +713,27 @@ public:
         AttrStrAtom *a = AttrStrAtom::toAtom(_a);
         if (a->linkcount!=(unsigned short)-1)
         {
+            unsigned table = a->hash % NUMTABLES;
+            CriticalBlock b(critk[table]);
             if (--(a->linkcount)==0)
             {
                 if (nc)
-                    htnc.remove((AttrStrNC *)a);
+                    htnc[table].remove((AttrStrNC *)a);
                 else
-                    htc.remove((AttrStrC *)a);
+                    htc[table].remove((AttrStrC *)a);
```

Comment:
minor: there could be a variety of remove that removed from the element from the HT, but returned it instead of destroying it.
Then destroy could be called outside of crit.